handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -746,7 +746,9 @@ public final SSLEngineResult wrap(
                         // We produced / consumed some data during the handshake, signal back to the caller.
                         // If there is a handshake exception and we have produced data, we should send the data before
                         // we allow handshake() to throw the handshake exception.
-                        return newResult(NEED_WRAP, 0, bytesProduced);
+                        //
+                        // unwrap(...) will then ensure we propagate the handshake error back to the user.
+                        return newResult(NEED_UNWRAP, 0, bytesProduced);
```

Comment:
cc @Scottmitch could I trouble you to check this out since it has your name on it :wink:

Reply:
@normanmaurer can we add a test for this?

Reply:
its covered by the tests already ;)